src/CoreBundle/private/modules/CMSRunCrons/CMSRunCrons.class.php

```diff
@@ -36,7 +37,7 @@ public function &Execute()
             if (!empty($sCronID)) {
                 $oTdbCmsCronJob = TdbCmsCronjobs::GetNewInstance();
                 $oTdbCmsCronJob->Load($sCronID);
-                if (false === $oTdbCmsCronJob->fieldLock) {
+                if (false === $oTdbCmsCronJob->fieldLock && true === $this->isCronjobsEnabled()) {
```

Comment:
I'd move the check to the begin of the method. If cronjobs are disabled, we avoid any side effects then (and improve performance insignificantly).

Maybe we should also notify the user that cronjobs are disabled, or they may keep hammering the execution button and wondering why nothing happens (maybe also add logging, so that unattended cronjob execution doesn't get lost unnoticed).

Reply:
It might change during execution. So this is checked very "locally".

Reply:
@ hammering: There are also other cases where there is no feedback. Ie if the cronjob is locked nothing at all (except for the popup) will happen.

Reply:
Logging for all (unattended) cronjobs improved.

Error messages shown for a single cronjob (started manually) if locked or disabled.

Reply:
Looks great, I'll add just some further minor remarks in separate conversations :-)